disable modp1 by default and deprecate modp2 and modp5 in getDiffieHellman

thinred opened the pull request:
Fixes #25366 by deprecating modp1, modp2 and modp5 groups in the docs. Some time should be given to the users to update their codebases.

mhdawson commented:
I think that since this would apply to 0.10.X stream we'd want a PR for merging the change into v0.10 and then we'd merge into 0.12.X.

The other question is whether we should add the check to crypto.createDiffieHellman(prime_length) as well.  ie reject a prime_length smaller than 768

thinred commented:
1st question: Ok, I'll rebase it.
2nd question: I'd say that no, the issue primarily applies to *well-known* groups (but, sure, using a small prime would be an issue, but it was an issue even before).

thinred committed:
Fix for the issue 25366.

In particular:
  - DH groups of size < 1024 are disabled by default
    (there is only one such group: modp1)
  - a new cmdline switch --enable-small-dh-groups and
    SMALL_DH_GROUPS_ENABLE env. variable are introduced;
    they override the default setting and therefore enable
    modp1 group
  - the docs & tests are updated

thinred commented:
I created a new PR on top of 0.10 in #25564.
